src/query/service/tests/it/sql/optimizer/heuristic/subquery.rs

```diff
@@ -99,6 +99,31 @@ pub async fn test_heuristic_optimizer_subquery() -> Result<()> {
             query: "select t.number from numbers(1) as t where exists(select * from numbers(1) as t1 where t.number > t1.number) and not exists(select * from numbers(1) as t1 where t.number < t1.number)".to_string(),
             rules: DEFAULT_REWRITE_RULES.clone(),
         },
+        Suite {
+            comment: "# Push down limit scan".to_string(),
+            query: "select * from (select t.number from numbers(10) as t limit 8) limit 9".to_string(),
+            rules: DEFAULT_REWRITE_RULES.clone(),
+        },
+        Suite {
+            comment: "# Push down limit scan".to_string(),
+            query: "select * from (select t.number from numbers(10) as t order by number desc) order by number asc".to_string(),
+            rules: DEFAULT_REWRITE_RULES.clone(),
+        },
+        Suite {
+            comment: "# Push down sort and limit to scan".to_string(),
+            query: "select number from (select t.number from numbers(10) as t order by number desc limit 8) order by number asc limit 9".to_string(),
+            rules: DEFAULT_REWRITE_RULES.clone(),
+        },
+        Suite {
+            comment: "# Push down limit join".to_string(),
+            query: "select t.number from numbers(1) as t, numbers(1) as t1 where t.number = (select count(*) from numbers(1) as t2, numbers(1) as t3 where t.number = t2.number) group by t.number order by t.number desc limit 3;".to_string(),
+            rules: DEFAULT_REWRITE_RULES.clone(),
+        },
+        Suite {
+            comment: "# Limit left join with complex expr".to_string(),
+            query: "select * from (select count(t1.number) as c1 from numbers(1) as t1 group by number) as t3 left join (select count(t.number) as c from numbers(2) as t group by number) as t4 on t3.c1=t4.c order by t3.c1 limit 1".to_string(),
+            rules: DEFAULT_REWRITE_RULES.clone(),
+        },
```

Comment:
The unit test will be migrated to logictest in #6643 , would you move them into `tests/logictest/suites/mode/standalone/limit.test`?

Reply:
absolutely ok~